prombench/Makefile

```diff
@@ -37,50 +30,96 @@ cluster_delete:
 		-v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
 		-f manifests/cluster_${PROVIDER}.yaml
 
+# /prombench <...> --bench.directory
+BENCHMARK_DIRECTORY := $(if $(BENCHMARK_DIRECTORY),$(BENCHMARK_DIRECTORY),manifests/prombench)
+# /prombench <...> --bench.version
+BENCHMARK_VERSION := $(if $(BENCHMARK_VERSION),$(BENCHMARK_VERSION),master)
+PROMBENCH_GIT_REPOSITORY ?= git@github.com:prometheus/test-infra.git
+PROMBENCH_DIR ?= .
+
+# maybe_pull_custom_version allows custom benchmarking as designed in
+# https://github.com/prometheus/proposals/pull/41. It allows calling
+# /prombench <release> --bench.version=<@commit or branch> which will cause
+# prombench GH job on Prometheus repo to call infra CLI with the non-master BENCHMARK_VERSION.
+# In such a case we pull a prombench repository for the given branch or commit version
+# and adjust PROMBENCH_DIR. As a result `make deploy` and `make clean` jobs
+# will apply /manifests/ apply custom manifests or even node pools.
+.PHONE: maybe_pull_custom_version
+maybe_pull_custom_version:
+ifeq (${BENCHMARK_VERSION},master)
+	@echo ">> Using standard benchmark configuration, from the docker image"
+else
+	@echo ">> Git pulling custom benchmark configuration from the ${BENCHMARK_VERSION}"
```

Comment:
```suggestion
	@echo ">> Git pulling custom benchmark configuration from the ${BENCHMARK_VERSION} branch"
```

nit

Reply:
It can also be commit at this point, so skipping - it will be clarified later (: